packages/core/src/index.ts

```diff
@@ -3,15 +3,17 @@ import dotenv from 'dotenv';
 import { findUp } from 'find-up';
 import Koa from 'koa';
 
-import initI18n from './i18n/init.js';
-
 dotenv.config({ path: await findUp('.env', {}) });
 
 // Import after env has configured
 
 const { loadConnectorFactories } = await import('./utils/connectors/factories.js');
 const { EnvSet } = await import('./env-set/index.js');
-const { tenantPool } = await import('./tenants/index.js');
+const { default: initI18n } = await import('./i18n/init.js');
+const [{ tenantPool }, { checkRowLevelSecurity }] = await Promise.all([
+  import('./tenants/index.js'),
+  import('./tenants/utils.js'),
+]);
```

Comment:
nit: Should we export `tenants/utils.js` from `tenants/index.js`? Then we don't have to write the `Promise.all` when importing them?

Reply:
i'm good with both

Reply:
i'm good with both